src/main/java/edu/ucsb/cs156/organic/controllers/CoursesController.java

```diff
@@ -132,4 +136,39 @@ public Iterable<Staff> getStaff(
         return courseStaff;
     }
 
+    @Operation(summary = "Update a course")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_INSTRUCTOR')")
+    @PutMapping("/update")
+    public Course updateCourse(
+            @Parameter(name = "courseId") @RequestParam Long courseId,
+            @RequestBody @Valid Course incoming) throws JsonProcessingException {
+        
+        User u = getCurrentUser().getUser();
+        Course course = courseRepository.findById(courseId)
+                .orElseThrow(() -> new EntityNotFoundException(Course.class, courseId.toString()));
+        Iterable<Staff> courseStaff = courseStaffRepository.findByCourseId(courseId);
+
+        //check if the user is a staff member of the course
```

Comment:
Ok, this is definitely not how we do things with a relational database!

You are doing a linear search through all of the course staff; but a SQL database is designed to do this for you.

I helped another student with this last Thursday; let me see if I can find that code and share it...

Reply:
Here's a PR with a suggestion: 

https://github.com/ucsb-cs156-f23/proj-organic-f23-6pm-1/pull/39

You can accept this PR into your branch if you are happy with the suggestion.